_source/_posts/2020-12-23-mulesoft-salesforce-integration.md

```diff
@@ -0,0 +1,650 @@
+
+u.localesidkey = 'en_US';
+
+u.emailEncodingKey = 'UTF-8';
+
+u.timeZoneSidKey = 'America/Los_Angeles';
+
+u.profileId = p.Id;
+
+insert u;
+
+return u;
+
+}
+
+global void updateUser(Id userId, Id portalId, Auth.UserData data){
+
+User u = new User(id=userId);
+
+u.email = data.email;
+
+u.lastName = data.lastName;
+
+u.firstName = data.firstName;
+
+}
+
+```
+
+Click *Save*.
+
+Navigate back to the *Apex Class* home page.
+
+Click *Security* on the **AutocreatedRegHandler**.
+
+Make sure you add *Standard Platform User* as an enabled **Profile**. Click *Save*.
+
+{% img blog/mulesoft-salesforce/image47.png alt:"Salesforce security" width:"800" %}{: .center-image }
+
+## Configure Trust over the remote URL called within Salesforce
+  
+Navigate to *Setup > Settings > Security > Remote Site Settings*
+  
+Click *New Remote Site*.
+
+Add your Okta URL (e.g.: `https://{yourDomainHere}.okta.com`) and click *Save & New*.
+
+{% img blog/mulesoft-salesforce/image48.png alt:"Salesforce new remote site" width:"800" %}{: .center-image }
+
+Add your *Salesforce Mulesoft Proxy Endpoint URL*. (e.g., `https://domain.cloudhub.io`)
+
+Click *Save*. You have now successfully added your Okta domain and Mulesoft Proxy Endpoint URL as trusted remote URLs.
+
+## Create a Visualforce page to tie everything together
+
+Within Salesforce, navigate to *Platform Tools > Custom Code > Visualforce Pages*
+
+Click *New*. Provide a **Label, Name, and Description**.
+
+{% img blog/mulesoft-salesforce/image49.png alt:"Salesforce new visualforce page" width:"800" %}{: .center-image }
+
+Generally, as per best practice, you would want to create a separate controller or APEX handler for your Visualforce custom page. For this tutorial, you'll re-use the same one you have modified earlier. Overwrite and paste the following code below:
+
+```html
+
+<!-- change the control value with the class name of your controller -->
+
+<apex:page controller="AutocreatedRegHandler1380796002690">
+
+<!-- Begin Default Content REMOVE THIS -->
+
+<h1>Congratulations</h1>
+
+This is your Salesforce page protected by Okta OIDC and OAuth 2.0
+
+<apex:form >
+
+<apex:commandButton id="btn" action="{!fetch_data}" value="Call Protected API behind Mulesoft using OAuth JWT Token below"/>
+
+<p/>
+
+Token: <apex:outputLabel id="one">{!accessToken}</apex:outputLabel>
+
+<p/>
+
+Access Token Details from https://yourdomain.oktapreview.com/oauth2/default/v1/userinfo: <p/>
+
+<apex:outputText >Sub : {!sub}</apex:outputText> <p />
+
+<apex:outputText >Full Name : {!name}</apex:outputText> <p />
+
+<apex:outputText >Email : {!email}</apex:outputText> <p />
+
+<apex:outputText >Username : {!userName}</apex:outputText> <p />
+
+<apex:outputText >First Name: {!firstName}</apex:outputText> <p />
+
+<apex:outputText >Last Name : {!lastName}</apex:outputText> <p />
+
+<apex:outputText >Callout : {!callOut}</apex:outputText>
+
+</apex:form>
+
+<!-- End Default Content REMOVE THIS -->
+
+</apex:page>
```

Comment:
```suggestion
```xml

<!-- change the control value with the class name of your controller -->
<apex:page controller="AutocreatedRegHandler1380796002690">

  <!-- Begin Default Content REMOVE THIS -->
  <h1>Congratulations</h1>
  
  This is your Salesforce page protected by Okta OIDC and OAuth 2.0
  
  <apex:form >
    <apex:commandButton id="btn" action="{!fetch_data}" value="Call Protected API behind Mulesoft using OAuth JWT Token below"/>
  
    <p/>
    Token: <apex:outputLabel id="one">{!accessToken}</apex:outputLabel>
    <p/>
  
    Access Token Details from https://yourdomain.oktapreview.com/oauth2/default/v1/userinfo: <p/>
  
    <apex:outputText >Sub : {!sub}</apex:outputText> <p />
    <apex:outputText >Full Name : {!name}</apex:outputText> <p />
    <apex:outputText >Email : {!email}</apex:outputText> <p />
    <apex:outputText >Username : {!userName}</apex:outputText> <p />
    <apex:outputText >First Name: {!firstName}</apex:outputText> <p />
    <apex:outputText >Last Name : {!lastName}</apex:outputText> <p />
    <apex:outputText >Callout : {!callOut}</apex:outputText>
  
  </apex:form>

<!-- End Default Content REMOVE THIS -->
</apex:page>
```

Just guessing here, but this would be some form of XML not HTML, and the extra whitespace _should_ be removed (and formated)